### PR TITLE
Fixed build error hal_rvv_071

### PR DESCRIPTION
**Pull request by MaximMilashchenko**
Fixed bug with enabling vector header when vector extension is disabled (RVV=OFF) in hal_rvv_071

```
force_builders=Custom
build_image:Custom=riscv-gcc
Xbuild_image:Custom=riscv-clang-rvv
Xbuild_image:Custom=riscv-clang-rvv-128
Xbuild_image:Custom=riscv-gcc-rvv-07
test_modules:Custom=core,imgproc,dnn
buildworker:Custom=linux-4
test_timeout:Custom=600
build_contrib:Custom=OFF
```

### CODE REVIEW

**Comment by opencv-alalek** [comment]
I believe we should disable RVV HAL instead.
There is no sense to keep it enabled without RVV compiler support.

**Comment by Junyan721113** [comment]
Currently NDSRVP is affected by the flag state.
CMake with Andes toolchains:
```shell
--   Other third-party libraries:
--     Lapack:                      NO
--     Custom HAL:                  YES (HAL RVV (ver 0.0.1) ndsrvp (ver 0.0.1))
--     Protobuf:                    build (3.19.1)
--     Flatbuffers:                 builtin/3rdparty (23.5.9)
```

```shell
In file included from /home/junyan/opencv_rvp/build/custom_hal.hpp:5,
                 from /home/junyan/opencv_rvp/modules/core/src/hal_replacement.hpp:968,
                 from /home/junyan/opencv_rvp/modules/core/src/precomp.hpp:92,
                 from /home/junyan/opencv_rvp/modules/core/src/alloc.cpp:43:
/home/junyan/opencv_rvp/3rdparty/hal_rvv/hal_rvv.hpp:8:10: fatal error: riscv_vector.h: No such file or directory
    8 | #include <riscv_vector.h>
      |          ^~~~~~~~~~~~~~~~
compilation terminated.
```

**Comment by opencv-alalek** [comment]
@Junyan721113 Please use `cmake ... -DWITH_HAL_RVV=OFF` as temporary workaround.

File: CMakeLists.txt [hunk 1]
```diff
@@ -265,7 +265,7 @@ OCV_OPTION(WITH_KLEIDICV "Use KleidiCV library for ARM platforms" OFF
   VISIBLE_IF (AARCH64 AND (ANDROID OR UNIX AND NOT IOS AND NOT XROS)))
 OCV_OPTION(WITH_NDSRVP "Use Andes RVP extension" (NOT CV_DISABLE_OPTIMIZATION)
   VISIBLE_IF RISCV)
-OCV_OPTION(WITH_HAL_RVV "Use HAL RVV optimizations" (NOT CV_DISABLE_OPTIMIZATION)
+OCV_OPTION(WITH_HAL_RVV "Use HAL RVV optimizations" ON
```
Comment (opencv-alalek) [inline CMakeLists.txt:268]: Revert this change. Keep `(NOT CV_DISABLE_OPTIMIZATION)`

Comment (MaximMilashchenko) [reply CMakeLists.txt]: Done

**Review by opencv-alalek — APPROVED** [review]
- [x] [Validated](http://pullrequest.opencv.org/buildbot/builders/precommit_custom_linux/builds/100445)
